### ISSUE TITLE
Launchpad Plugins Installation from GAMA 1.8 RC2 doesn't work

### ISSUE THREAD

**Issue by agrignard** [gama-platform/gama#2550]
**Describe the bug**
Try to install a launchpad plugin 

**To Reproduce**
1. Go to Help->Install New Plugins
2. Choose work with http://updates.gama-platform.org/experimental/oxygen
3. Choose Launchpad

**Expected behavior**
Normal installation

**Actual Behavior**
Every thing seems fine until GAMA asks to restart. 
<img width="1918" alt="screen shot 2018-09-13 at 15 28 16" src="https://user-images.githubusercontent.com/3928502/45510929-a9061f80-b769-11e8-8aaf-a04e3d0d8041.png">

The restart never happens , it stops after asking for workspace information and then there is no way to see GAMA even in the task manager(the only way it too delete the current.app and redownload a new one).

**Desktop (please complete the following information):**
 - Mac os 10.11
 - GAMA 1.8 RC2

**Comment by agrignard** [comment]
... ok I just test it with another plugins like unity it works...

So maybe it is only happening for launchpad?

**Comment by benoitgaudou** [comment]
Hi,

I just tried the launchpad plugin with GAMA Git version and I get an error:

****************************************************************************************************


The initialization of GAML artefacts went wrong. If you use the developer
version, please clean and recompile all plugins.

Otherwise post an issue at https://github.com/gama-platform/gama/issues


****************************************************************************************************



Error in loading EventLayerStatement delegate : Plug-in
"mit.gaml.extension.launchpad" was unable to instantiate class
"mit.gaml.extension.launchpad.skills.LaunchPadEventLayer".

Have you check that you plugin work ?

Cheers

Benoit


Le jeu. 13 sept. 2018 à 21:36, Arnaud Grignard <notifications@github.com> a
écrit :

**Comment by hqnghi88** [comment]
I post a fix, but this should be in the issue list of experimental repository.

**Comment by AlexisDrogoul** [comment]
Yep, you're right. This issue should not be here

**Comment by agrignard** [comment]
I have a new issue while trying to install the launchpad plugin on the official 1.8 release.

The plugins is well installed howver I get a compiel error when writting this


`````
event "pad_down" type: "launchpad" action: updateGrid;
``````

Saying
````
No event can be triggered for 'pad_down'. Acceptable values are
   [mouse_exit, mouse_enter, mouse_move, mouse_down, mouse_up]
   or a character
````

**Comment by agrignard** [comment]
Here is the model that is supposed to work

````
/**
* Name: Launch Pad Event Feature
* Author: Arnaud Grignard 
* Description: Model which shows how to use the event layer to trigger an action with a LaunchPad Novation (This model only work with the launchpad plugins extension available in GAMA 1.7 since January 2018)
* Tags: tangible interface, gui, launchpad
 */
model event_layer_model

global skills:[launchpadskill]
{
	list<string> buttonColors <-["red","orange","brown","yellow","lightyellow","green","darkgreen","black"];
	map<string,string> function_map <-["UP"::buttonColors[0],"DOWN"::buttonColors[1],"LEFT"::buttonColors[2],"RIGHT"::buttonColors[3],"SESSION"::buttonColors[4],"USER_1"::buttonColors[5],"USER_2"::buttonColors[6],"MIXER"::buttonColors[7]];
	init{
	  do resetPad;
	  do setButtonLight colors:buttonColors;	
	}
	
	action updateGrid
	{   
		if(function_map.keys contains buttonPressed and buttonPressed != "MIXER"){
		    ask cell[ int(padPressed.y *8 + padPressed.x)]{color <- rgb(function_map[buttonPressed]);}
		    do setPadLight color:function_map[buttonPressed];
		}
		if(buttonPressed = "MIXER"){
			ask cell[ int(padPressed.y *8 + padPressed.x)]{color <- #white;}
		}			
		if(buttonPressed="ARM"){
			do resetPad;
			do setButtonLight colors:buttonColors;	
			ask cell{
				color<-#white;
			}
		}
		do updateDisplay;
	}	
}

grid cell width: 8 height: 8;

experiment Displays type: gui
{
	output
	{
		display View_change_color 
		{
			grid cell lines: #black;
			event "pad_down" type: "launchpad" action: updateGrid;
		}
	}
}
````
